src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -243,6 +248,32 @@ public override async Task<CompletionChange> GetChangeAsync(Document document, C
             return CompletionChange.Create(textChange);
         }
 
+        private static async Task<TextSpan> GetTextChangeSpanAsync(Document document, TextSpan startSpan, CancellationToken cancellationToken)
+        {
+            var result = startSpan;
+            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(result.Start);
+            if (syntaxFactsService.IsStringLiteral(token) || syntaxFactsService.IsVerbatimStringLiteral(token))
+            {
+                result = token.Span;
+                var contentStartsAt = token.Text.IndexOf('"');
+                contentStartsAt = contentStartsAt >= 0
+                    ? contentStartsAt + 1
+                    : contentStartsAt;
+                var contentEndsAt = token.Span.Length == 0
+                    ? token.Span.End
+                    : token.Text[token.Span.Length - 1] == '"'
+                        ? token.Span.End - 1
+                        : startSpan.End; // The string is open ended. We keep the right end as is.
```

Comment:
do you have tests for this case?

Reply:
Yes. A few. Including one that's failing for now (See also my comment in the thread). I think the best would be to extend the span to the right until the end of line or some whitespace. What do you think.

Reply:
Yes.  that woudl fit how the normal "compute the text span" logic is intended to work.


In fact, to make your life easier.  Just call: CommonCompletionUtilities.GetWordSpan.  However, for the isWordCharacter helper pass: `ch => SyntaxFacts.IsIdentifierPartCharacter(ch) || ch == '.'`

Reply:
This doesn't work if you there are escape sequences before the cursor.

Reply:
Do we ever expect there to be escape characters in an assembly string like this?  I highly doubt it.  And, as such, i think it's fine to have the simple code that works for the 99.99% case :)

Reply:
Note: if you do want to support escapes, why not just add \ into the list of acceptable characters to consume?